File: python/14. Logest Common Prefix.py
# https://leetcode.com/problems/longest-common-prefix/

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = ""
        pointer = 0

        while True:
            str = strs[0]
            if pointer >= len(str):
                break
            char = str[pointer]
            for s in strs[1:]:
                if pointer >= len(s) or char != s[pointer]:
                    char = ""
                    break
            
            if char == "":
                break
            
            prefix += char
            pointer += 1
        
        return prefix

File: swift/14. Logest Common Prefix.swift
// https://leetcode.com/problems/longest-common-prefix/

class Solution {
    func longestCommonPrefix(_ strs: [String]) -> String {
        let mappedStrs = strs.map(Array.init)

        var prefix: [Character] = []
        var pointer: Int = 0

        let str = mappedStrs.first!
        while pointer < str.count {
            guard pointer < str.count else { break }

            var allMatch = true
            let char = str[pointer]
            for s in mappedStrs[1...] {
                if pointer >= s.count || char != s[pointer] {
                    allMatch = false
                    break
                }
            }

            if !allMatch {
                break
            }

            prefix.append(char)
            pointer += 1
        }

        return String(prefix)
    }
}